src/backend/dx12ll/src/factory.rs

```diff
@@ -231,6 +260,21 @@ impl core::Factory<R> for Factory {
         }
     }
 
+    fn create_buffer_heap(&mut self, heap_type: &core::HeapType, size: u64) -> native::Heap {
+        self.create_flagged_heap(heap_type, size, 
+                                 winapi::D3D12_HEAP_FLAGS(0) | winapi::D3D12_HEAP_FLAG_DENY_NON_RT_DS_TEXTURES | winapi::D3D12_HEAP_FLAG_DENY_RT_DS_TEXTURES)

```

Comment:
There are aliases which should be used here instead of manually assembling the flags:
* `D3D12_HEAP_FLAG_ALLOW_ONLY_BUFFERS`
* `D3D12_HEAP_FLAG_ALLOW_ONLY_NON_RT_DS_TEXTURES`
* `D3D12_HEAP_FLAG_ALLOW_ONLY_RT_DS_TEXTURES`